File: dart/maroctax/lib/src/cnss.dart
import 'fiscal_config.dart';
import 'types.dart';
import 'utils.dart';

/// Employee CNSS contribution (4.48% of gross, capped monthly).
double calculateCnss(double gross, [FiscalConfig? config]) {
  final cfg = config ?? getFiscalConfig();
  final base = gross < cfg.cnssEmployeeCap ? gross : cfg.cnssEmployeeCap;
  return roundMoney(base * cfg.cnssEmployeeRate);
}

File: typescript/maroctax/src/cnss.ts
import type { FiscalConfig } from './types.js';
import { roundMoney } from './utils.js';

/**
 * Employee CNSS contribution (4.48% of gross, capped monthly).
 * @param gross - Monthly gross salary in MAD.
 * @param config - Fiscal configuration for the payroll year.
 */
export function calculateCnss(gross: number, config: FiscalConfig): number {
  const base = Math.min(gross, config.cnssEmployeeCap);
  return roundMoney(base * config.cnssEmployeeRate);
}
